src/main/java/io/warp10/script/ext/influxdb/HttpClientUtils.java

```diff
@@ -0,0 +1,60 @@
+//
+//   Copyright 2021  SenX S.A.S.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.script.ext.influxdb;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import okhttp3.OkHttpClient.Builder;
+
+import io.warp10.script.WarpScriptException;
+
+public class HttpClientUtils {
+
+    private static final String KEY_READ_TIMEOUT = "readTimeout";
+    private static final String KEY_WRITE_TIMEOUT = "writeTimeout";
+
+    public static Builder getOkHttpClientBuilder(final String functionName, final Map<Object, Object> params) throws WarpScriptException {
+
+        Builder result = new Builder();
+
+        Long readTimeout = getLongValueFromKey(functionName, params, KEY_READ_TIMEOUT, (long) result.getReadTimeout$okhttp());
+        result.readTimeout(readTimeout, TimeUnit.MILLISECONDS);
+
+        Long writeTimeout = getLongValueFromKey(functionName, params, KEY_WRITE_TIMEOUT, (long) result.getWriteTimeout$okhttp());
+        result.writeTimeout(writeTimeout, TimeUnit.MILLISECONDS);
+
+        return result;
+    }
+
+    private static Long getLongValueFromKey(final String functionName, final Map<Object, Object> params, final String key, final Long defaultValue) throws WarpScriptException{
```

Comment:
The method could simply return a primitive long and the defaultValue parameter could also be a primitive long. The use of the method could simply be result.readtimeout(getLongValueFromKey(functionName, params, KEY_READ_TIMEOUT, (long) result.getReadTimeout$okhttp())), same with result.writeTimeout.